### PR TITLE
18538 Resulting Business Name and Type component (WIP)

### PR DESCRIPTION
**Pull request by jamespaologarcia**
*Issue #:* bcgov/entity#18538

Still need to fix 1 failing unit test, and add new unit tests for the added features. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/common/HelpText.vue
Comment (severinbeauvais) [inline src/components/common/HelpText.vue:2]: Our convention is that the top element (which isn't template) holds the id of this component.

In this case, it would be "help-text", but that's very generic and you've put this component in the common folder. Please move it to common/Amalgamation folder and rename this more specifically in case we have other help text components.

File: src/components/common/HelpText.vue
Comment (severinbeauvais) [inline src/components/common/HelpText.vue:25]: Also by convention, rename this to match the id (and filename).

File: src/components/common/HelpText.vue
Comment (severinbeauvais) [inline src/components/common/HelpText.vue:6]: Ask Yui if he wants to use HTML entity `&sect;` (§) instead of "s.".

File: src/components/common/HelpText.vue
Comment (severinbeauvais) [inline src/components/common/HelpText.vue:22]: Blank line above this, please.

File: src/components/common/NameRequestInfo.vue
Comment (severinbeauvais) [inline src/components/common/NameRequestInfo.vue:113]: This is a common component, so your changes here affect other filings.

I don't fancy the idea of a duplicate component for amalgamation filings, and I don't love the idea of adding `v-if="isAmalgamationFiling"` in a common component (as this adds special cases), but I think that will be easiest and cleanest way to go for now.

File: tests/unit/NameRequestInfo.spec.ts
Comment (severinbeauvais) [inline tests/unit/NameRequestInfo.spec.ts:256]: Since you will NOT be changing this component for everybody (ie, you'll have specific changes for amalgamation filings), then you'll need to add some unit tests in which you've set the filing type.

**Review by severinbeauvais — CHANGES_REQUESTED** [review]
Looks really good overall. See comments.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-591-2bknaabm.web.app

**Comment by severinbeauvais** [comment]
The help component works but the formatting is different from the uxpin. Check the layout and font colours.

image

File: src/components/common/NameRequestInfo.vue
Comment (severinbeauvais) [inline src/components/common/NameRequestInfo.vue:1]: Add a case to `requestType` on line 197+ to display "Amalgamation".

File: src/components/common/HelpText.vue
Comment (yuisotozaki) [reply src/components/common/HelpText.vue:6]: "s." is good. Thanks!

File: src/components/common/HelpText.vue
Comment (yuisotozaki) [reply src/components/common/HelpText.vue:15]: Please use 250-387-7848.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-591-2bknaabm.web.app

File: src/components/common/NameRequestInfo.vue
Comment (severinbeauvais) [inline src/components/common/NameRequestInfo.vue:157]: Is this a continuation of the v-else-if on line 100? If so, then THIS block will be rendered if "this is not an amalgamation filing" or "there is a name request number". This logic seems incorrect; should THIS code block be `v-else-if="!getNameRequestNumber"` ?

Another way to resolve this is to test this component with a named amalgamation, a numbered amalgamation, a named non-amalgamation (eg, BC IA) and a numbered non-amalgamtion.

File: tests/unit/NameRequestInfo.spec.ts
Comment (severinbeauvais) [inline tests/unit/NameRequestInfo.spec.ts:258]: ^^ beautiful!

Also please add a blank line between the previous block and this new describe block.

File: tests/unit/NameRequestInfo.spec.ts
Comment (severinbeauvais) [inline tests/unit/NameRequestInfo.spec.ts]: So you've changed what's rendered here for registrations (and presumably IAs also). Was this approved by Yui?

Is this a new common section for registrations + IAs + amalgamations?

File: tests/unit/NameRequestInfo.spec.ts
Comment (JazzarKarim) [inline tests/unit/NameRequestInfo.spec.ts:218]: Can you please add a blank line above too please?

File: tests/unit/NameRequestInfo.spec.ts
Comment (jamespaologarcia) [reply tests/unit/NameRequestInfo.spec.ts]: I added this back on the latest push

File: tests/unit/NameRequestInfo.spec.ts
Comment (yuisotozaki) [reply tests/unit/NameRequestInfo.spec.ts]: Yes - this is OK. The discrepancies are from amalgamation design having started several months ago and a lot has changed since then.

**Comment by severinbeauvais** [comment]
Sample BC Limited numbered amalg for you (try it yourself on preview site or locally):

image

File: tests/unit/NameRequestInfo.spec.ts
Comment (severinbeauvais) [reply tests/unit/NameRequestInfo.spec.ts]: Before we resolve this conversation, @jamespaologarcia , could you please show a numbered IA in dev vs a numbered IA in preview site, for Yui to also approve?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-591-2bknaabm.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-591-2bknaabm.web.app

**Comment by severinbeauvais** [comment]
Somehow, the bullets have disappeared.

From https://business-create-dev--pr-591-2bknaabm.web.app/amalg-reg-information?id=TzWiPgtz8N:

image

From https://dev.create.business.bcregistry.gov.bc.ca/incorporation-define-company?id=TIXK6nbssn&accountid=3040:

image

I think this is a uxpin issue... @yuisotozaki, which way do you want it?

**Review by severinbeauvais — APPROVED** [review]
Looks good.

~Wait for Yui's response on the bullets. He may want them back. Or, we could merge this PR and you could create another if the bullets need to be fixed. Thoughts?~ _bullets are now fixed_

**Comment by jamespaologarcia** [comment]
> Looks good.
> 
> Wait for Yui's response on the bullets. He may want them back. Or, we could merge this PR and you could create another if the bullets need to be fixed. Thoughts?

I think it's better to keep it in this ticket since adding bullets won't be a major change

File: src/components/common/NameRequestInfo.vue
Comment (severinbeauvais) [inline src/components/common/NameRequestInfo.vue]: I see a handful of GitHubActions (lint) warnings below. Can you fix please?

Comment (jamespaologarcia) [reply src/components/common/NameRequestInfo.vue]: I updated this part and omitted the changes I've made with the latest push

File: src/components/common/NameRequestInfo.vue
Comment (severinbeauvais) [reply src/components/common/NameRequestInfo.vue:157]: I'm expecting you (James) to have tested the following 4 scenarios (and compared with Dev where appropriate):
1. named non-amalgamation (eg, IA or registration)
2. numbered non-amalgamation
3. named amalgamation
4. numbered amalgamation

I'm resolving this conversation. Please unresolve it (or start a new conversation) if you find anything amiss.

**Review by severinbeauvais — APPROVED** [review]
Still LGTM but please get another review approval or 2 before merging.

**Review by leodube-aot — APPROVED** [review]

**Pull request merged**
